pypdf/filters.py

```diff
@@ -285,7 +285,8 @@ def decode(
         index = 0
         while True:
             if index >= len(data):
-                raise PdfStreamError("Unexpected EOD in ASCIIHexDecode")
+                break  # reach End Of String even if no EOD
```

Comment:
Does this have any side-effect? Should we add a warning / a debug log message?

Reply:
My analysis is that this should not been an issue. If prefered I add a warning